internal/indexmanagement/reconcile.go

```diff
@@ -140,65 +134,47 @@ func ReconcileCurationConfigmap(apiclient client.Client, cluster *apis.Elasticse
 	})
 	return errCtx.Wrap(err, "failed to update configmap")
 }
-
-func ReconcileRolloverCronjob(apiclient client.Client, cluster *apis.Elasticsearch, policy apis.IndexManagementPolicySpec, mapping apis.IndexManagementPolicyMappingSpec, primaryShards int32) error {
-	if policy.Phases.Hot == nil {
-		log.Info("Skipping rollover cronjob for policymapping; hot phase not defined", "policymapping", mapping.Name)
+func ReconcileIndexManagementCronjob(apiclient client.Client, cluster *apis.Elasticsearch, policy apis.IndexManagementPolicySpec, mapping apis.IndexManagementPolicyMappingSpec, primaryShards int32) error {
+	containers := []*corev1.Container{}
+	if policy.Phases.Delete != nil {
 		return nil
-	}
-	schedule, err := crontabScheduleFor(policy.PollInterval)
-	if err != nil {
-		return kverrors.Wrap(err, "failed to reconcile rollover cronjob", "policymapping", mapping.Name)
-	}
-	conditions := calculateConditions(policy, primaryShards)
-	name := fmt.Sprintf("%s-rollover-%s", cluster.Name, mapping.Name)
-	payload, err := json.Marshal(map[string]rolloverConditions{"conditions": conditions})
-	if err != nil {
-		return kverrors.Wrap(err, "failed to serialize the rollover conditions to JSON")
-	}
-	envvars := []corev1.EnvVar{
-		{Name: "PAYLOAD", Value: base64.StdEncoding.EncodeToString(payload)},
-		{Name: "POLICY_MAPPING", Value: mapping.Name},
-	}
-	fnContainerHandler := func(container *corev1.Container) {
-		container.Command = []string{"bash"}
-		container.Args = []string{
-			"-c",
-			"/tmp/scripts/rollover",
+		minAgeMillis, err := calculateMillisForTimeUnit(policy.Phases.Delete.MinAge)
+		if err != nil {
+			return err
 		}
+		envvars := []corev1.EnvVar{
+			{Name: "POLICY_MAPPING", Value: mapping.Name},
+			{Name: "MIN_AGE", Value: strconv.FormatUint(minAgeMillis, 10)},
+		}
+		containers = append(containers, newInitContainer(cluster.Name, "delete", constants.PackagedElasticsearchImage(), "/tmp/scripts/delete", envvars))
+	} else {
+		log.V(1).Info("Skipping curation cronjob for policymapping; delete phase not defined", "policymapping", mapping.Name)
```

Comment:
this seems misleading... its that we aren't configuring for delete, right?

Reply:
updating message to try and be more clear... `log.V(1).Info("Skipping curation management for policymapping; delete phase not defined", "policymapping", mapping.Name)`